Source/WebCore/fileapi/Blob.cpp

```diff
@@ -383,4 +383,9 @@ WebCoreOpaqueRoot root(Blob* blob)
     return WebCoreOpaqueRoot { blob };
 }
 
+size_t Blob::memoryCost() const
+{
+    return size();
```

Comment:
I am not sure it is great to use size(): it is using sync IPC to network process.
Maybe we should instead report back from network process the actual size asynchronously?
Also, it is not really thread safe if memoryCost is not called in Blob ScriptExecutionContext thread.

Also, I am unclear whether Blob size is correct for memortCost.
In case of new Blob([chunk1, chunk2]), this is pretty clearly a good approximation.
In case of new Blob([blob1, blob2]), the blob does not trigger any additional memory in network process (past blob1 and blob2), but we would still consider its size is blob1.size + blob2.size.
Is that good enough for memory cost?

Also what about File that inherits from Blob?

Reply:
For what it's worth, this is what MessageEvent::memoryCost() is relying on already.

Reply:
> In case of new Blob([blob1, blob2]), the blob does not trigger any additional memory in network process (past blob1 and blob2), but we would still consider its size is blob1.size + blob2.size.
Is that good enough for memory cost?

memoryCost() is just a hint AFAIK, so it doesn't have to be 100% accurate. Also, I think reporting `blob1.size + blob2.size` is actually the right thing here. Otherwise, when the blob1 and blob2 wrappers get garbage collected, then the memory cost of the ` Blob([blob1, blob2])` would no longer be accurate (since no longer 0).

Reply:
> Also, it is not really thread safe if memoryCost is not called in Blob ScriptExecutionContext thread.
I assume the context thread is paused when called from a GC thread. Otherwise, our existing memoryCost() functions would be unsafe too.

Reply:
> Also what about File that inherits from Blob?

It is true that if the Blob is a File only backed by a file on disk, then the memory cost is not accurate. It would return the size of the file on disk :/

Reply:
> I am not sure it is great to use size(): it is using sync IPC to network process.

True, but only the first time though and only when we didn't know the size upon Blob construction, then the size is cached inside Blob.

Reply:
I think that when the File object is a bundle, `size()` would also trigger creating a zip archive from it.

Reply:
I believe the value returned by memoryCost() can change.

Reply:
For sliced data, I believe returning the size of the slice is the right thing to do either way (whether data is in its own memory or still refers to the original blob).